nvflare/fuel/f3/cellnet/cell.py

```diff
@@ -155,31 +226,36 @@ def send_request(self, channel, target, topic, request, timeout=10.0, secure=Fal
                 optional=optional,
             )
 
-        encode_payload(request, StreamHeaderKey.PAYLOAD_ENCODING)
+        try:
+            encode_payload(request, StreamHeaderKey.PAYLOAD_ENCODING)
+        except BaseException as exc:
+            self.logger.error(f"Can't encode {request=} {exc=}")
 
         req_id = str(uuid.uuid4())
         request.add_headers({StreamHeaderKey.STREAM_REQ_ID: req_id})
 
         # this future can be used to check sending progress, but not for checking return blob
-        future = self.send_blob(channel, topic, target, request, secure)
+        self.logger.debug(f"{req_id=}: About to send_blob")
+        future = self.send_blob(channel=channel, topic=topic, target=target, message=request, secure=secure)
 
         waiter = SimpleWaiter(req_id=req_id, result=make_reply(ReturnCode.TIMEOUT))
         self.requests_dict[req_id] = waiter
+        self.logger.debug(f"{req_id=}: Waiting starts")
 
         # Three stages, sending, waiting for receiving first byte, receiving
 
         # sending with progress timeout
         self.logger.debug(f"{req_id=}: entering sending wait {timeout=}")
         sending_complete = self._future_wait(future, timeout)
         if not sending_complete:
-            self.logger.debug(f"{req_id=}: sending timeout")
+            self.logger.info(f"{req_id=}: sending timeout")
             return self._get_result(req_id)
         self.logger.debug(f"{req_id=}: sending complete")
 
         # waiting for receiving first byte
         self.logger.debug(f"{req_id=}: entering remote process wait {timeout=}")
         if not waiter.in_receiving.wait(timeout):
-            self.logger.debug(f"{req_id=}: remote processing timeout")
+            self.logger.info(f"{req_id=}: remote processing timeout")
```

Comment:
Include value of "timeout" in the log msg.

Reply:
Added in the new commit